### ISSUE TITLE
[YSQL] SQL Copy From command error with primary/foreign key tables

### ISSUE THREAD

**Issue by phendric-uk** [yugabyte/yugabyte-db#12199]
### Description

When using the copy command to load a table with a foreign key constraint, the error message:

ERROR:  insert or update on table "employee" violates foreign key constraint "employee_dept_id_fkey"
DETAIL:  Key (dept_id)=(2) is not present in table "dept".

is shown.

Since the value of 2 is in the primary key constraint table column, this message should not be happening.  
It still completely loads the data in the file into the table with the foreign key constraint issue regardless of this error.

Using test tables:
dept - has the primary key constraint
employee - has the foreign key constraint

![copy error](https://user-images.githubusercontent.com/96289556/163999820-7d689f2f-4493-4b66-97ec-a59c04d21275.jpg)

The above screenshot shows the table definition and the error

**Comment by bllewell** [comment]
### Summary

I reproduced this issue. I also demonstrated that the same test runs without error in vanilla Postgresql. See below.

### Detail

**Preamble**

Please always state the YB version and o/s env where you see an issue.

I tested using YB-2.13.0.1 on MacOS Big Sur Version 11.6.5. I also tested on PostgreSQL Version 14.1—on the same MacOS.

It helps to submit a machine readable (set of) scripts(s). This makes diagnosis far easier and less prone to copy-and-paste error than presenting just a screenshot. I wrote scripts to re-capture your test. I tested them on PG and on YB. Everything works as expected using PG. But the error manifests in YB. I simplified your case just a little by using simple test files for _'copy'_ with _space_ as the delimiter. (So I had to use single-word names.)

Notice that with a case like yours, where the functionality is supposed to be the same on PG as on YB, it's a very good plan to run your tests in both PG and YB.

- Sometimes (but not in this case) the test fails in vanilla PG because of a straight user error. Testing without error in PG first makes the analysis clearer.
- Very rarely (but not in this case) the test fails in PG because of a genuine PG bug. YB will then simply inherit this bug.

- Sometimes, things work as expected in PG but fail in YB because of a bug in YB's adoption of the PG SQL processing code. This is the case with the present issue.

**bllewell's testcase—first using vanilla PG**

First, do this generic setup so that the test can be run, time and again, without relying on pre-existing artifacts:

```
\c postgres postgres
set client_min_messages = warning;
drop database if exists db;
create database db owner postgres;

\c db postgres
set client_min_messages = warning;
drop schema if exists public cascade;
create schema s1 authorization postgres;
set schema 's1';
```
Notice that I created and used a dedicated scheme _'s1'_ rather than relying on _'public'_. This is a better practice. Notice too, that I'm using the user/role _postgres_ because _'copy'_ from a server-side o/s file needs a superuser.

Next, do this testcase-specific setup:

```
create table dept(id int primary key, name text);
create table employee(
  id int primary key,
  dept_id int,
  name text,
  constraint employee_fk foreign key(dept_id) references dept(id));

prepare qry as
select
  p.id as "dept.id",
  p.name as "dept.name",
  c.id as "employee.id",
  c.name as "employee.name"
from dept p inner join employee c on c.dept_id = p.id
order by p.id, c.id;

insert into dept(id, name) values (1, 'sales'), (2, 'marketing');
insert into employee(id, dept_id, name) values(1, 1, 'fred');
execute qry;
```

This runs without error and the query brings this result:

```
 dept.id | dept.name | employee.id | employee.name 
---------+-----------+-------------+---------------
       1 | sales     |           1 | fred
```

Next, create two text files thus:

_t1.txt_
```
2 1 adam
3 2 joe
```
_t2.txt_
```
4 3 bill
```

Do this positive test:

```
copy employee(id, dept_id, name) from '/Users/Bllewell/Desktop/issue-12199/t1.txt' delimiter ' ';
execute qry;
```

The _'copy'_ completes without error and brings this result:

```
 dept.id | dept.name | employee.id | employee.name 
---------+-----------+-------------+---------------
       1 | sales     |           1 | fred
       1 | sales     |           2 | adam
       2 | marketing |           3 | joe
```

Now, do this negative test:

```
copy employee(id, dept_id, name) from '/Users/Bllewell/Desktop/issue-12199/t2.txt' delimiter ' ';
```

It fails (as expected) on PG thus:

```
ERROR:  insert or update on table "employee" violates foreign key constraint "employee_fk"
DETAIL:  Key (dept_id)=(3) is not present in table "dept".
```

And _'execute qry'_ (of course) shows that nothing as changed.

**bllewell's testcase—now using -2.13.0.1**

The two parts of the setup run without error in YB and bring the same result. But the (supposedly) positive test (using _t1.txt_) now fails thus—just as _phendric-uk_ reported:

```
ERROR:  insert or update on table "employee" violates foreign key constraint "employee_fk"
DETAIL:  Key (dept_id)=(1) is not present in table "dept".
```

The negative test (using _t2.txt_) also fails—but this time the failure is expected—thus:

```
ERROR:  insert or update on table "employee" violates foreign key constraint "employee_fk"
DETAIL:  Key (dept_id)=(3) is not present in table "dept".
```

**Comment by lnguyen-yugabyte** [comment]
@phendric-uk @bllewell : It should be a duplicate to #10594 , which has been fixed in 2.13.2.0 (and backported to 2.12).

**Comment by paullee-yb** [comment]
Fixed by the following [commit](https://github.com/yugabyte/yugabyte-db/commit/7c428da49ce9009df8faf9788c53c9c71cb7178b).
